src/backend/distributed/utils/foreign_constraint_utils.c

```diff
@@ -0,0 +1,369 @@
+/*-------------------------------------------------------------------------
+ *
+ * foreign_key_utils.c
```

Comment:
I don't like when we put stuff in the utils directory that implements core database logic. 

Maybe it's time we introduce a DDL directory?

Reply:
Yes, this would be good. 
Would you like me to create a ddl directory in the same level with planner, executor, connection etc.? 

Additionally, what should we put in it besides foreign_key_utils.c? If we are going to find and relocate every DDL related functions under this newly created directory, maybe I should do this in another PR.